core/src/com/unciv/ui/civilopedia/CivilopediaText.kt

```diff
@@ -111,3 +177,106 @@ object MarkupRenderer {
         return table.apply { pack() }
     }
 }
+
+/** Storage class for interface [ICivilopediaText] for use as base class */
+open class CivilopediaText : ICivilopediaText {
+    override var civilopediaText = listOf<FormattedLine>()
+}
+/** Storage class for instantiation of the simplest form containing only the lines collection */
+class SimpleCivilopediaText(lines: List<FormattedLine>, val isComplete: Boolean = false) : CivilopediaText() {
+    init {
+        civilopediaText = lines
+    }
+    override fun hasCivilopediaTextLines() = true
+    override fun replacesCivilopediaDescription() = isComplete
+    constructor(strings: Sequence<String>, isComplete: Boolean = false) : this(
+        strings.map { FormattedLine(it) }.toList(), isComplete)
+    constructor(first: Sequence<FormattedLine>, strings: Sequence<String>, isComplete: Boolean = false) : this(
+        (first + strings.map { FormattedLine(it) }).toList(), isComplete)
```

Comment:
why .toList() after the .map{}? You're getting a sequence, which is already good?

Reply:
We have `constructor(List<FormattedLine>, Boolean)` and  `constructor(Sequence<String>, Boolean)` up to that point, not a `constructor(Sequence<FormattedLine>, Boolean)` - trivial to write but would be single-use and it would toList() anyway. The final storage is a List, int's not abstract, because elsewhere the superclass wants to be able to have that easily deserialized from json.

I think that's a 'no that kind of sequence is not good yet'